libs/pageserver_api/src/shard.rs

```diff
@@ -209,6 +292,74 @@ impl<'de> Deserialize<'de> for TenantShardId {
     }
 }
 
+impl Serialize for ShardIndex {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
```

Comment:
Sweet, I didn't know that existed.

However, where do we actually do the binary serialization? 
I think for `IndexPart::layer_metadata`, it's `is_human_readable() == true`, no?

At a minum, please add an explainer comment on the `impl Serialize for ShardIndex` that justifies this complexity.